src/types/routeMethods.ts

```diff
@@ -6,14 +6,14 @@ import { Query } from '@/utilities/query'
 
 export type RouteMethods<
   TRoutes extends Routes = Routes,
-  TPathParams extends Record<string, unknown> = Record<never, never>,
-  TQueryParams extends Record<string, unknown> = Record<never, never>
-> = Identity<UnionToIntersection<RouteMethodsTuple<TRoutes, TPathParams, TQueryParams>[number]>>
+  TPathParams extends Record<string, unknown[]> = Record<never, never>,
+  TQueryParams extends Record<string, unknown[]> = Record<never, never>
+> = UnionToIntersection<RouteMethodsTuple<TRoutes, TPathParams, TQueryParams>[number]>
```

Comment:
Removing this `Identity` means
- Non public routes show up in the object as `never`. 
- Typescript can defer evaluation of nested types until they are access. Which gives us a huge performance boost.